Raw-loader setup for templates and refactoring of connectionCtrl

pfiaux opened the pull request:
I wanted this to be a pull request for #18 but I forgot to refer to it in the commit so here's a separate pull request.
## 1. Raw loader setup

I setup raw loader and integrated with with webpack so after `npm install` it should work again.
- From the javascript file we can require templates with `require('./templatePath/templateName.tpl.html')`
- The extension for template is configured to `.tpl.html`

This will allow breaking up index.html into smaller pieces.
## 2. connectionCtrl refactoring

The recommended way to break up the code is with components (controller and template together). I wasn't able easily add a template for connectionCtrl because it's inside of statCtrl and it has profileCtrl inside of it.

So what I did is remove the connectionCtrl:
- The version was moved to its own component
  - I also created a `versionService` so any other controller can also get it if needed
- I moved the `connection` status to the ProfileCtrl which was displaying it
- `isLoaded` was not used, removed it.
  Then the connectionCtrl was empty and it could be removed.
## 3. statsTotal component

I did another example of how to move parts of the index.html indo it's own components by refactoring the div displaying the total into a component as well.

pfiaux committed:
Added and configured the raw loader, refactored version and stats total as components with separate templates.

The refactoring made the connection controller redundant it only set a status variable, moved it to profileCtrl and removed the connectionCtrl.

pfiaux commented:
@Cyclodex can you review the changes? Let me know if you have questions about the template, raw-loader or angular components.

I did some testing but i'm not familiar with all the new features so you might want to double check it before you merge it, there should be no change for the user only the way the code is structured.

Cyclodex commented:
This just sound awesome, thanks for the contribution. I will have to check it and fix the conflict.

I am sorry for the controller in controller part, i did this in the last feature update, because I did not know how to make it better. Will check what you did and try to have them separated.
The issue was that I wanted to display 2 different things in the footer, i am sure we can improve the situation here.

Cyclodex commented on webpack.config.js:
```diff
@@ -25,8 +25,13 @@ module.exports = {
+                test: /\.tpl\.html$/,
+                loader: 'raw-loader',
+                exclude: /node_modules/
+            },
```
You are the pro about webpack, but I am wondering , what is that doing now here ?

pfiaux replied:
It tells webpack:
- If a file that ends in `.tpl.html` is required
- Use the raw-loader to process it (because it's not javascript)
- Except if the file/require is in `node_modules/` then ignore it

So basically every time you use `require('somePath/someTemplate.tpl.html')` in angular webpack will replace it with the template when it builds the app.

The one above was a little bit different because It has to run less so that we have a css file and then the extract plugin moves it to a separate file otherwise it would inline it with the javascript.

Cyclodex commented on js/stats/stats.controller.js:
```diff
@@ -1,3 +1,7 @@
+// Require sub components and services
+require('./statsTotal.component.js');
+require('../version/version.component.js'); // version is currently inside stats
```
Does this mean, we should load the version component somewhere else? Or where / how would you implement this? I think it doesn't need to be part of stats... Its something globally kind of... Is probably confusing when we load it here in the stats...

pfiaux replied:
My idea was since it's only used in the stats then if you don't use the stats it wont be loaded, but that's probably over optimization :) but you're right it's a global component, moving it to the cyclone.js file would be less confusing.

If you want to change the setup you could refactor have footer component (or controller) which mostly does the layout and in the footer you can load the version, the stats, and so on. Then the version will be even more independent from the stats.

Commit:
BIG changes, trying to make a footer component which holds the footer.
Further splitting up will happen later, lets keep it working.

Stats controller is gone, will need to split up the stats output in several components and a service or similar I guess... (both components need the same calculation-base).
Profile Ctrl can also be changed later into a component. Also there we need the USER in different places, so its rather something else what we can inject everywhere.
Directive?Factory?Service?

refs #26

Commit:
@pfiaux did I understand this correctly. Instead of $scope I use $ctrl now in here.
Can you please review if that makes sense ? Thanks :)

refs #26

Commit:
Merge branch 'master' into setup-raw-loader
Solving following conflicts:

# Conflicts:
#	js/time/stats.controller.js
#	js/user/connection.controller.js

Cyclodex commented:
@pfiaux please review my changes. I tried to start doing some change on the footer.
There is still lots to do, but is it going into the right direction?
- check my change from `$scope` to `$ctrl` - is this correct?
- check my code commits above, would be great to hear some feedback, thanks in advance!

(**PS** I also merged from master to solve the conflicts on this branch)._(as suggested by github)_

Commit:
Merge branch 'master' into setup-raw-loader
Solving following conflicts:

refs #28

# Conflicts:
#	js/time/stats.controller.js
#	js/user/connection.controller.js

pfiaux commented:
I looked over the changes, yes you want to use `$ctrl` with components. Otherwise it looks good.

For the stats the plan in your comment sounds good with 2 components, for the part that has the same base you can put that into a service and then use it in both components (for example first you move the common code into 1 function like `var statsData = parseStatsData(rawData)` and second you move that function into a service).

For the ProfileCtrl it's more complicated because it initializes some login stuff on load (`window.addEventListener...`) and it stores the user in the root scope.
So first I would move that into a profileService or userService (You can use it to store the user and get it anywhere including in the ProfileCtrl). Probably you want to use an `angular.module('Cyclone').run(...)` in cyclone.js to call the `userService.initialize()` instead of the load event listener. And make sure the app can run well like that.
Then the ProfileCtrl will probably only get the user from the new userSerivce and display it, at that point it should be much easier to convert it into a component.

When I say services I would do it using factory() function for example [Data services](https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#data-services) might be a good starting point.

Commit:
$ctrl is ok, lets move on here.
Further changes can be made later, want to push this into master now.

refs #26
